tests/experimental/datasources/test_utils.py

```diff
@@ -0,0 +1,377 @@
+        },
+        "expect_column_max_to_be_between": {
+            "value": 6,
+            "rendered_template": "$column maximum value must be greater than or equal to $min_value and less than or equal to $max_value.",
+        },
+        "expect_column_median_to_be_between": {
+            "value": 1,
+            "rendered_template": "$column median must be greater than or equal to $min_value and less than or equal to $max_value.",
+        },
+    }
+    assert len(validation_result.results) == 3
+
+    for r in validation_result.results:
+        assert r.success
+        assert (
+            r.result["observed_value"]
+            == expected_metric_values[r.expectation_config.expectation_type]["value"]
+        )
+
+        if include_rendered_content:
+            # There is a prescriptive atomic renderer on r.expectation_config
+            num_prescriptive_renderer = len(r.expectation_config.rendered_content)
+            assert (
+                num_prescriptive_renderer == 1
+            ), f"Expected exactly 1 rendered content, found {num_prescriptive_renderer}"
+            rendered_content = r.expectation_config.rendered_content[0]
+            assert rendered_content.name == AtomicPrescriptiveRendererType.SUMMARY
+            assert (
+                rendered_content.value.template
+                == expected_metric_values[r.expectation_config.expectation_type][
+                    "rendered_template"
+                ]
+            )
+
+            # There is a diagnostic atomic renderer on r, a validation result result.
+            num_diagnostic_render = len(r.rendered_content)
+            assert (
+                num_diagnostic_render == 1
+            ), f"Expected 1 diagnostic renderer, found {num_diagnostic_render}"
+            diagnostic_renderer = r.rendered_content[0]
+            assert (
+                diagnostic_renderer.name == AtomicDiagnosticRendererType.OBSERVED_VALUE
+            )
+            assert (
+                diagnostic_renderer.value.schema["type"]
+                == "com.superconductive.rendered.string"
+            )
+        else:
+            assert r.rendered_content is None
+            assert r.expectation_config.rendered_content is None
+
+    # Rudimentary test for data doc generation
+    docs_dict = context.build_data_docs()
+    assert "local_site" in docs_dict, "build_data_docs returned dictionary has changed"
+    assert docs_dict["local_site"].startswith(
+        "file://"
+    ), "build_data_docs returns file path in unexpected form"
+    path = docs_dict["local_site"][7:]
+    with open(path) as f:
+        data_doc_index = f.read()
+
+    # Checking for ge-success-icon tests the result table was generated and it was populated with a successful run.
+    assert "ge-success-icon" in data_doc_index
+    assert "ge-failed-icon" not in data_doc_index
```

Comment:
Is all of this logic necessary? This feels like so much to assert against - what is the purpose of our tests? Can we narrow our scope at all so we don't have brittle tests?

Reply:
@cdkini This is not new (came from before Spark).  However, I believe that it is temporary and will be replaced with a more fully-featured and performant implementation soon.  Thanks.

Reply:
This should be our e2e test so the goal here is to test that we can go from a zep datasource and generate datadocs and they look correct (ie we see no expectations failing, metrics look correct, the datadocs look right, etc). It is a lot of stuff but I wanted something that was comprehensive since this was a new entry point to this workflow.
The goal was to have 1 e2e test for each major workflow (computing expectations, checkpoints, data assistants) and then lots of unit tests for particular functionality.